Modules/Sources/DesignSystem/Foundation/IconName.swift

```diff
@@ -0,0 +1,32 @@
+import UIKit
+import SwiftUI
+
+/// `Icon` provides a namespace for icon identifiers.
+///
+/// The naming convention follows the SF Symbols guidelines, enhancing consistency and readability.
+/// Each icon name is a dot-syntax representation of the icon's hierarchy and style.
+///
+/// For example, `ellipsis.vertical` represents a vertical ellipsis icon.
+public enum IconName: String, CaseIterable {
+    case ellipsisHorizontal = "ellipsis.horizontal"
```

Comment:
It might be worth reading out to the infra team and asking them to integrate [SwiftGen](https://github.com/SwiftGen/SwiftGen) that can generate these automatically. It will not only reduce the amount of manual work but will provide build-time checks for the availability of resources.

Btw, what does the `DS` mean?

Reply:
> It might be worth reading out to the infra team and asking them to integrate [SwiftGen](https://github.com/SwiftGen/SwiftGen) that can generate the similar code for accessing assets automatically.

Agreed, what do you think @mokagio 

> Btw, what does the `DS` mean?

It stands for Design System ✨ 

We wanted to namespace this design system so it doesn't conflict with the old one.

Reply:
@salimbraksa @kean I think integrating SwiftGet is a great idea :)